### DISCUSSION TITLE
Understanding Hysterisis

### DISCUSSION THREAD

**Discussion by vjbnmrdg** [Rem0o/FanControl.Releases#2315] [General]
Could you please explain how the hysterisis function works with the following 3 scenarios?

Scenario 1: Hysterisis set to 0 degrees and response time to 1 second.

Scenario 2: Hysterisis set to 2 degrees and response time to 1 second.

Scenario 3: Hysterisis set to 1 degree and response time to 10 seconds. 

Ultimately, the 3 scenarios will allow me to wrap my head around the functionality but I am trying to avoid the scenario of temperature creep (very slow temperature increases over a prolonged period of time). If hysterisis is set to 1 degree and 1 second, what if the temperature increases by 1 degree every few minutes? Will the fans ever increase or could this lead to the system overheating? I understand this is an unlikely scenario, but its something I can potentially encounter in my particular use case.

I have my step up and step down times dialed in, so I don't believe there will be any problem with using the most responsive hysterisis settings.

Thank you!

**Comment by Rem0o** [comment]
Hysteresis is stateful, meaning it remembers at what temperature it last changed.

Given an initial temperature of 40 deg.

Scenario 1: It will trigger for every temperature change ( internal clock is 1hz, so 1 second is basically no delay)
Scenario 2: It will trigger as soon as 38 or 42 degree is hit.
Scenario 3: It will trigger if during 10 __consecutive__ seconds the temperature is at 41 and up or 39 and down.

So for scenario 3 with that slow creep, it would trigger 10 seconds after it hit that 41 degree.

**Reply by vjbnmrdg** [reply]
@Rem0o - Awesome. Thanks so much. That makes sense.  

Have you seen any negative affects from a scenario 1 set up? As in, is triggering for every temperature change more resource intensive and/or reduce fan lifespan?

**Reply by Rem0o** [reply]
Scenario 1 is totally unnecessary and overkill. Plus the changing of speed all the time is very audibly annoying.
